### PR TITLE
feat: adding all icons from the design system to the codebase

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

This PR adds ALL of the icons from the SIP-34 (#8976)  design system into the codebase. Many were already here, but under different names, or with different attributes (sizes, etc). Now they're all here, visible in storybook, and existing use cases have been updated to work with updated icons wherever relevant.

Forthcoming steps/PRs:
• Removing any additional hard-coded colors in the SVGs, replacing them with a series of overridable default colors based on Theme variables
• Replacing font-awesome icons with these new icons wherever possible

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/stylesheets/superset.less [hunk 1]
```diff
@@ -529,17 +529,6 @@ tr.reactable-column-header th.reactable-header-sortable {
   padding-right: 17px;
 }
 
-.fave-unfave-icon {
```
Comment (rusackas) [inline superset-frontend/stylesheets/superset.less:532]: this... doesn't seem to really do anything.

File: superset-frontend/src/components/NavDropdown/index.tsx [hunk 2]
```diff
@@ -33,16 +33,16 @@ const NavDropdown = styled(ReactBootstrapNavDropdown)`
 
   :after {
     content: '';
-    height: ${({ theme }) => theme.gridUnit}px;
-    width: ${({ theme }) => theme.gridUnit * 2}px;
-    background: url('/static/assets/images/icons/dropdown-arrow.svg');
```
Comment (rusackas) [inline superset-frontend/src/components/NavDropdown/index.tsx:38]: the dropdown-arrow was small. The triangle-down button is bigger, with padding around it. These changes accommodate the layout change.

File: superset-frontend/src/explore/main.less [hunk 3]
```diff
@@ -36,7 +36,6 @@
   margin-bottom: 40px;
 }
 
-.fave-unfave-icon,
```
Comment (rusackas) [inline superset-frontend/src/explore/main.less:39]: this doesn't appear necessary these days.

**Comment by zhouyao1994** [comment]
it's great~

**Comment by mistercrunch** [comment]
OMG yes!

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**